cpp/src/arrow/adapters/orc/adapter_test.cc

```diff
@@ -157,4 +225,2478 @@ TEST(TestAdapter, readIntAndStringFileMultipleStripes) {
     EXPECT_TRUE(stripe_reader->ReadNext(&record_batch).ok());
   }
 }
+
+// WriteORC tests
+
+// General
+TEST(TestAdapterWriteGeneral, writeZeroRows) {
+  std::vector<std::shared_ptr<Field>> xFields{field("bool", boolean()),
```

Comment:
What is the reason that you use slightly different schemas in every test? Maybe you can generalize this to have a reusable test data generation.

Reply:
These tests are mostly amalgamated from the 31,000+ lines of ORC test that used to exist. Overall if we generate random test data we should be able to generate random Arrow tables and use them in ORC tests as well.